cllc-public-app/Controllers/EstablishmentsController.cs

```diff
@@ -153,24 +165,144 @@ private List<EstablishmentMapData> GetMapData(string search)
+                    }
+
+                    establishmentMapData = new List<EstablishmentMapData>();
+                    if (licences != null)
+                    {
+                        foreach (var license in licences)
+                        {
+                            if (license.AdoxioLicenceType != null && license.AdoxioLicenceType.AdoxioName.Equals("Licensee Retail Store")
+                                                                  && license._adoxioEstablishmentValue != null)
+                            {
+                                var establishment = license.AdoxioEstablishment;
+
+                                if (establishment != null &&
+                                    (establishment.AdoxioLicencee == null ||
+                                     establishment.AdoxioLicencee.Name != LDB_ACCOUNT_NAME) &&
+                                    establishment.AdoxioLatitude != null && establishment.AdoxioLongitude != null)
+                                {
+
+                                    establishmentMapData.Add(new EstablishmentMapData
+                                    {
+                                        id = establishment.AdoxioEstablishmentid,
+                                        Name = establishment.AdoxioName,
+                                        License = license.AdoxioLicencenumber,
+                                        Phone = establishment.AdoxioPhone,
+                                        AddressCity = establishment.AdoxioAddresscity,
+                                        AddressPostal = establishment.AdoxioAddresspostalcode,
+                                        AddressStreet = establishment.AdoxioAddressstreet,
+                                        Latitude = (decimal) establishment.AdoxioLatitude,
+                                        Longitude = (decimal) establishment.AdoxioLongitude,
+                                        IsOpen = establishment.AdoxioIsopen.HasValue && establishment.AdoxioIsopen.Value
+                                    });
+                                }
+                            }
+                        }
                     }
+                    var cacheEntryOptions = new MemoryCacheEntryOptions()
+                               // Set the cache to expire in an hour.                   
+                               .SetAbsoluteExpiration(TimeSpan.FromHours(1));
+
+                    // Save data in cache.
+                    _cache.Set("S_" + cacheKey, establishmentMapData, cacheEntryOptions);
+                    cacheEntryOptions = new MemoryCacheEntryOptions()
+                               // Set the cache to expire in an hour.                   
+                               .SetAbsoluteExpiration(TimeSpan.FromDays(1));
+                    // long term cache
+                    _cache.Set(cacheKey, establishmentMapData, cacheEntryOptions);
```

Comment:
remove this line as it is redundant

Reply:
Turns out it was required, however set the long term cache to 2 days and the short term to 1 day